code/modules/mob/living/say.dm

```diff
@@ -242,13 +242,10 @@ proc/get_radio_key_from_channel(var/channel)
 
 		//DO NOT FUCKING CHANGE THIS TO GET_OBJ_OR_MOB_AND_BULLSHIT() -- Hugs and Kisses ~Ccomp
 		var/list/hear = hear(message_range,T)
-		var/list/hearturfs = list()
 
-		for(var/I in hear)
-			if(ismob(I))
-				var/mob/M = I
-				listening += M
-				hearturfs += M.locs[1]
+		for(var/mob/M in mob_list)
+			if(M && M.locs.len && M.locs[1] in hear)
```

Comment:
`M` can not be `null`. 
`for(var/TYPE/VAR in LIST)`  implicitly call `istype(VAR, TYPE)`